docs/cow-protocol/tutorials/cow-swap/flash-loans.mdx

```diff
@@ -0,0 +1,119 @@
+
+The order itself intends to:
+1. Request a flash loan to Aave of 5,000 USDT.
+2. Use those 5,000 to partially repay the 10,000 USDT debt on Aave.
+3. Withdraw 400,000 USDC from Aave.
+4. Perform a trade on CoW Swap, selling up to 400,000 USDC for 5,002.5 USDT (borrowed amount + [0.05% flash loan fee on Aaave](https://aave.com/docs/concepts/flash-loans)).
+2. Use the 5,002.5 USDT from the trade to fully repay the flash loan (including the fee).
+
+:::info
+
+Liquidity and exchange ratios on the Sepolia network can be very inconsistent. To ensure successful execution of the order, we use a high amount of USDC (both as collateral and as the token being traded).
+
+General considerations when testing flash loans on Sepolia and Aave:
+- [Token names in Aaave may not correspond](https://github.com/mrnaveira/cow-flashloan-tester?tab=readme-ov-file#2-select-aave-tokens) with tokens names in other dapps.
+- [Token ratios](https://github.com/mrnaveira/cow-flashloan-tester?tab=readme-ov-file#4-determine-token-amounts-needed-to-trade-and-deposit) may need to be balanced, and trading amounts adjusted accordingly.
+- Tokens in Aave can be [obtained from the faucet](https://github.com/mrnaveira/cow-flashloan-tester?tab=readme-ov-file#5-get-tokens-from-the-faucet).
+- Check [withdrawal and borrowing availability](https://github.com/mrnaveira/cow-flashloan-tester?tab=readme-ov-file#7-check-aaves-withdrawalborrow-availability) of the tokens on Aave.
+- Try a different RPC in case of transaction submission issues.
+
+In a production network (e.g. mainnet), these considerations generally won’t apply.
+
+:::
+
+### Order construction
+
+```json5
+{
+  "from": "Safe address",
+  "sellToken": "USDC address", // collateral token (unlocked by repaying the debt)
+  "buyToken": "USDT address", // originally borrowed token that was now advanced by the flash loan
+  "buyAmount": 5002500000, // appData.flashloan.amount + 0.05% flashloan fee
+  "receiver": "settlementContract", // this is for repaying the flash loan
+  "validTo": "now + 5m", // managing risk can be done by having a short validity
+  "kind": "buy", // buy exactly the flash loaned amount and keep the surplus in the collateral token
+  "partiallyFillable": false, // if an order is partially fillable, then it is not ensured the debt will be paid
+  "appData": {
+    "hooks": {
+      "pre": [
+         // First: Repay the debt
+         {
+            "target": "0x123...321",
+            "value": "0",
+            "callData": "signedRepayCall"
+         },
+         // Then: Withdraw the collateral
+         {
+            "target": "0x123...321",
+            "value": "0",
+            "callData": "signedWithdrawCall"
+         }
+      ]
+    },
+    "flashloan": {
+      "token": "USDT",
+      "amount": 5000000000 // 5,000 USDT in atoms
+    }
+  }
+}
+```
+
+Check the [flash loan tester script](https://github.com/mrnaveira/cow-flashloan-tester) for more details on how this order is constructed.
```

Comment:
Seems a bit weird to have a dedicated tutorial page and then defer to a repo. Why not copy the respective code parts into the docs itself and explain each step a bit more?
It's not a great example but I was thinking about sth like [this](https://fasterthanli.me/articles/catching-up-with-async-rust) for example.

Reply:
I reworked the tutorial to expand all step as suggested